src/backend/distributed/utils/colocation_utils.c

```diff
@@ -112,6 +112,9 @@ get_colocated_shard_array(PG_FUNCTION_ARGS)
 	Oid arrayTypeId = OIDOID;
 	int colocatedShardIndex = 0;
 
+	/* sort to get consistent output */
+	colocatedShardList = SortList(colocatedShardList, CompareShardIntervalsById);
```

Comment:
Heh. Is this necessary to do _here_? Or can it be done in SQL in the tests?

Reply:
Could be done with doing `UNNEST` & `ORDER BY`. But this seemed to cover wider queries with a single touch.